src/types/inc/Utf8Utf16Convert.hpp

```diff
@@ -1,34 +1,31 @@
-/*++
+﻿/*++
 Copyright (c) Microsoft Corporation
 Licensed under the MIT license.
 
 Module Name:
-- UTF8OutPipeReader.hpp
+- Utf8Utf16Convert.hpp
 
 Abstract:
-- This reads a UTF-8 stream and gives back a buffer that contains complete code points only
-- Partial UTF-8 code points at the end of the buffer read are cached and prepended to the next chunk read
+- Defines functions for converting between UTF-8 and UTF-16 strings.
+- Defines classes to complement partial code points at the begin of a string and cache partials from the end of a string.
+- Defines classes to do both the complement/cache task and the conversion task at once.
 
 Author(s):
-- Steffen Illhardt (german-one) 12-July-2019
+- Steffen Illhardt (german-one) 2019
 --*/
 
 #pragma once
 
-#ifndef WIN32_LEAN_AND_MEAN
-#define WIN32_LEAN_AND_MEAN
-#endif
+[[nodiscard]] HRESULT U8ToU16(_In_ const std::string_view u8Str, _Out_ std::wstring& u16Str, bool discardInvalids = false) noexcept;

```

Comment:
I'm hoping we can offer both an HRESULT-return-code pattern like this one and also a second variant of these functions that returns the wstring directly (and throws exceptions) such that it can fit nicely into the codebase in both the result-code-comparing circumstances AND the ones that are moving to more modern exception based stuff and want to `const` the resultant string right away.

Reply:
I hope I understood your requests correctly.